### ISSUE TITLE
[BUG] Repackaging App with Adaptive Icons shows Stock Android Icon

### ISSUE THREAD

**Issue by kyuds** [iBotPeaches/Apktool#3619]
### Information
1. **Apktool Version (`apktool -version`)** - 2.9.3
2. **Operating System (Mac, Linux, Windows)** - macOS (Intel Chip)
3. **APK From? (Playstore, ROM, Other)** - Other (cannot share, unfortunately)
4. **Java Version (`java --version`)** - 17

So first of all, thank you for the wonderful project. I am currently in the process of editing an APK, and I've noticed that everything works fine EXCEPT for the application icon shown on the home screen (and app menu). Rebuilding the APK and installing it allows for the app to work fine, (including login and rendering), but the icon appears as the stock, green-ish background android icon.

After some inspection, I've noticed that this particular icon uses Adaptive Icons with the two layers, and observing that the decoded android manifest file, points to a completely different link (launcher_icon instead of ic_launcher), it seems that Adaptive Icons seems to be an issue. 

I have already tried unpacking and repacking the APK without changing anything (except setting the extractNativeLibs field to true, as my APK is a split), and still the icon is not working properly.

Thank you!

**Comment by iBotPeaches** [comment]
Do you have a small sample of the AndroidManifest file decoded, then built/decoded again? To see the icon section.

**Comment by kyuds** [comment]
Honestly don't know if this will help, but:
base.apk: AndroidManifest.xml:
```
<application android:appComponentFactory="androidx.core.app.CoreComponentFactory" android:extractNativeLibs="true" android:icon="@mipmap/launcher_icon" android:isSplitRequired="true" android:label="@string/app_name" android:name="<removed name>" android:requestLegacyExternalStorage="true" android:usesCleartextTraffic="false">
```
base.apk: The launcher_icon.xml in res/mipmap-anydpi-v26
```
<?xml version="1.0" encoding="utf-8"?>
<adaptive-icon
  xmlns:android="http://schemas.android.com/apk/res/android">
    <background android:drawable="@null" />
    <foreground android:drawable="@null" />
</adaptive-icon>
```
The foreground and background images for the launcher icon is in split_config.xxhdpi.apk, but looking through them I wasn't able find any file similar to the "launcher_icon.xml" file in base.apk.

Also, I tried switching out the "@null" to the appropriate paths for the fore/background images, but that also didn't work.

**Comment by iBotPeaches** [comment]
Thanks - So I'm guessing the adaptive icons are in the split application? Since I don't see how Apktool would null out the drawable refs unless the resource is not valid in that specific application.

**Comment by kyuds** [comment]
Yup. The adaptive icons are in the split apk (split_config.xxhdpi.apk). Im starting to think that it is because the application is built with Flutter

**Comment by AndnixSH** [comment]
I'm getting same issue with Cửu Âm Chân Kinh 2D Mobile (https://apkcombo.com/id/cuu-am-chan-kinh-2d-mobile/vn.gosu.cuuam2d/). The game works, just no icon

1. Extract base APK (vn.gosu.cuuam2d.apk) from XAPK
2. Decompile and compile without modification
3. Put compiled vn.gosu.cuuam2d.apk back in XAPK
4. Install XAPK using 3rd party apps like XAPK Installer https://play.google.com/store/apps/details?id=com.wuliang.xapkinstaller&hl=en

Here is the AndroidManifest.xml sample

[AndroidManifest.txt](https://github.com/user-attachments/files/20983067/AndroidManifest.txt)

**Comment by sipsuru** [comment]
Seems related to #3806 

https://github.com/iBotPeaches/Apktool/issues/3806#issuecomment-2720768205

Have you tried merging split apks?

**Comment by AndnixSH** [comment]
> Seems related to [#3806](https://github.com/iBotPeaches/Apktool/issues/3806)
> 
> [#3806 (comment)](https://github.com/iBotPeaches/Apktool/issues/3806#issuecomment-2720768205)
> 
> Have you tried merging split apks?

Yes, but the game would get stuck on black screen. But nevermind, I found a fix by hooking game lib
